api/api.go

```diff
@@ -461,6 +523,159 @@ func CreateOrb(ctx context.Context, logger *logger.Logger, namespace string, nam
 	return orb, err
 }
 
+func incOrbVersion(orbVersion string, segment string) (string, error) {
+	v, err := semver.NewVersion(orbVersion)
+	if err != nil {
+		return "", err
+	}
+
+	var v2 semver.Version
+	switch segment {
+	case "major":
+		v2 = v.IncMajor()
+	case "minor":
+		v2 = v.IncMinor()
+	case "patch":
+		v2 = v.IncPatch()
+	}
+
+	return v2.String(), nil
+}
+
+// IncrementOrb accepts an orb and segment to increment the orb.
+func IncrementOrb(ctx context.Context, logger *logger.Logger, configPath string, namespace string, orb string, segment string) (string, error) {
+	name := namespace + "/" + orb
+	id, err := getOrbID(ctx, logger, name)
+	if err != nil {
+		return "", err
+	}
+
+	v, err := OrbVersion(ctx, logger, namespace, orb)
+	if err != nil {
+		return "", err
+	}
+
+	v2, err := incOrbVersion(v, segment)
+	if err != nil {
+		return "", err
+	}
+
+	response, err := OrbPublishID(ctx, logger, configPath, id, v2)
```

Comment:
I believe this is racy and could end up clobbering a previous version.

Is it possible to add the previous version to the query so that it will only update when the previous version matches?

Reply:
This is indeed racy, as we're getting the latest version in a separate request before publishing a new one. If someone else where to publish a version in-between those requests... :boom: 

I'm unsure how to solve this without somehow modifying the API to perform the update atomically in a single request?

Reply:
One way is to accept the current version as a parameter in the graphQL. The API server can perform an `UPDATE ... WHERE` using the existing version. If the version doesn't match (because some other process updated it), then the query will update 0 rows, and the API server can return an error.